src/tool/xmeta/cpp/ast_to_st_listener.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "generated/XlangParserBaseListener.h"
+#include "symbol_table.h"
+
+#include <string_view>
+
+using xlang::xmeta::xmeta_symbol_table;
+
+struct xlang::xmeta::enum_model;
+struct xlang::xmeta::enum_member;
+
+struct ast_to_st_listener : XlangParserBaseListener
+{
+    ast_to_st_listener() = delete;
+    ast_to_st_listener(xmeta_symbol_table & symbol_table);
+
+    void exitEnum_declaration(XlangParser::Enum_declarationContext *ctx) override;
+
+    void enterNamespace_declaration(XlangParser::Namespace_declarationContext *ctx) override;
+    void exitNamespace_declaration(XlangParser::Namespace_declarationContext *ctx) override;
+
+private:
+    xmeta_symbol_table & m_st;
```

Comment:
Possible lifetime concern here with this type holding a raw reference. On its own, it's not evil, but all the usage I'm seeing is creating these two objects in the raw on their own, so things could get a little sketchy, especially with how verbose all the usage is so far.

Is there a wrapper/helper that could improve things twofold by 1) making lifetime guarantees by ensuring that the listener is a local that will go out of scope (and so it won't outlive the symbol table) and 2) hiding details that aren't needed for the end user. In my dream world, I wouldn't have to deal with the lexer, parser, tree, or any of the other types when invoking this.